src/coreclr/tests/src/JIT/opt/InstructionCombining/CondToBranchlessExpr.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+public class Program
+{
+    public static int returnCode = 100;
+
+    public static int Main(string[] args)
+    {
+        var testsWithOpt = new Tests();
+        var testsWithoutOpt = new TestsWithoutOptimization();
+
+        var allTestsWithOpt = testsWithOpt.GetType().GetMethods()
+            .Where(m => m.Name.StartsWith("Test"))
+            .OrderBy(m => m.Name)
+            .ToArray();
+
+        var allTestsWithoutOpt = testsWithoutOpt.GetType().GetMethods()
+            .Where(m => m.Name.StartsWith("Test"))
+            .OrderBy(m => m.Name)
+            .ToArray();
+
+        int[] testInput = Enumerable.Range(-100, 100)
+            .Concat(new [] {int.MaxValue - 1, int.MaxValue})
+            .ToArray();
+
+        for (int i = 0; i < allTestsWithOpt.Length; i++)
+        {
+            for (int j = 0; j < testInput.Length; j++)
+            {
+                var invokeArgs = new object[] { testInput[j] };
+                int actual = (int)allTestsWithOpt[i].Invoke(testsWithOpt, invokeArgs);
+                int expected = (int)allTestsWithoutOpt[i].Invoke(testsWithoutOpt, invokeArgs);
+
+                if (actual != expected)
+                {
+                    Console.WriteLine($"{allTestsWithOpt[i].Name}: {actual} != {expected}");
+                    returnCode++;
+                }
+            }
+        }
+
+        return returnCode;
+    }
+}
+
+
+public class Tests
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static T Cns<T>(T t) => t; // don't want Roslyn to optimize anything
+
+    public int Test1(int a) => a == 42 ? Cns(0) : Cns(1);
```

Comment:
>a == 42 ? [](start = 31, length = 9)

The IL we get for if-else might be similar as that of tertiary operator but for completeness should we also add test with simple if-else?
```
if (a == 42) {
 return 0;
} else {
 return 1;
}
```